flink/flink1.12-shims/src/main/java/org/apache/zeppelin/flink/Flink112Shims.java

```diff
@@ -173,13 +173,26 @@ public Object createCatalogManager(Object config) {
 
   @Override
   public String getPyFlinkPythonPath(Properties properties) throws IOException {
+    if ("yarn_application".equalsIgnoreCase(properties.getProperty("flink.execution.mode"))) {
+      // for yarn application mode, FLINK_HOME is container working directory
+      String flinkHome = new File(".").getAbsolutePath();
+      List<File> depFiles = null;
+      depFiles = Arrays.asList(new File(flinkHome + "/lib/python").listFiles());
+      StringBuilder builder = new StringBuilder();
+      for (File file : depFiles) {
+        LOGGER.info("Adding extracted file {} to PYTHONPATH", file.getAbsolutePath());
+        builder.append(file.getAbsolutePath() + ":");
+      }
+      return builder.toString();
```

Comment:
It is almost the same as the 191~198 line code, so it would be good to separate it with a function.

Reply:
Good point, Fixed